### PR TITLE
Adding modal pattern. closes #321

### LINKED ISSUE #321: Develop modal pattern
Related to and blocks #307. Also see #307 for mockups. Related to 18F/web-design-standards#911

### PR DESCRIPTION
**Pull request by webinista**
- Adds app/assets/stylesheets/_scss/caseflow-components/_modals.scss partial
- Adds JavaScript pattern for removing hidden attribute, closing modal, updating location
- Adds _modals.scss partial to contain modal styles
- Adds modal example to start.html.erb screen

`_modal.html.erb` uses `raw` to print HTML contained in `modal_html`. I believe this is safe as long as it is never used with untrusted/user-entered data. There is a comment to this effect in the file.

Also switched to `<a href="#confirm-cancel-certification" class="cf-action-openmodal cf-btn-link">Can't certify</a>` instead of `<%= link_to "Can't certify", :not_certified, class: 'cf-btn-link pull-left' %>` in `start.html.erb` because the _Can't certify_ button link is now a modal trigger, rather than a link to the error page. The error page will instead be linked from another modal.

### CODE REVIEW

**Comment by paultag** [comment]
The `start` action appears to do a noop back to the same page -- can we link to the error page?

**Comment by paultag** [comment]
(Ooops, sorry for the close)

**Comment by webinista** [comment]
> The start action appears to do a noop back to the same page -- can we link to the error page? 

@paultag Do you mean the _Can't certify_ link? That's by design. That link functions as a trigger that opens the modal. The error page will be linked from the  _Skip_ button on the certification feedback screen (shown below; ticket is #340). I assume that whichever form handler processes the _Submit Feedback_ button (see #341) can also redirect to the error page.

Since #340 builds heavily on this pattern, I wanted have this approach (specifically the `raw` html output) vetted before continuing. 

cannotcertify_02_feedback

File: app/views/web/start.html.erb [hunk 1]
```diff
@@ -62,11 +62,20 @@
 
     <div class="usa-cf_app-segment">
 
-    <%= link_to "Can't certify", :not_certified, class: 'cf-btn-link pull-left' %>
+    <a href="#confirm-cancel-certification" class="cf-action-openmodal cf-btn-link">Can't certify</a>
```
Comment (paultag) [inline app/views/web/start.html.erb]: Can we change all of `Can't certify` to `Cancel Certification`? 

(from @KHarshawat)

Comment (webinista) [inline app/views/web/start.html.erb]: I'm tracking that in #334.

**Comment by paultag** [comment]
Awesome, thanks!

**Pull request merged**